packages/moda-sdk/src/core/ModaClient.ts

```diff
@@ -0,0 +1,37 @@
+import { PublicClient, WalletClient } from 'viem';
+import { createProfile, createProfileFor } from './profile';
+import { CreateProfileArgs } from './types';
+
+export class ModaClient {
```

Comment:
What about using a functional approach without classes? I find they are more composable and easier to test

Reply:
Should i maybe use a config to do this kind of thing?

``
setConfig({
  publicClient, 
  walletClient
})
``
and in the methods

``
const {publicClient, walletClient} = getConfig()
``

Reply:
I went with this, it's cleaner I think.